### ISSUE TITLE
out of the box support for URLs

### ISSUE THREAD

**Issue by FunkyMuse** [Lavmee/precompose-navigation-typesafe#9]
if we have something like

```
@Serializable
    data class ImagePreview(val url : String) : AppDestinations
```

and let's sasy that the url we pass is `https://fastly.picsum.photos/id/866/200/300.jpg?hmac=rcadCENKh4rD6MAp6V_ma-AyWv641M4iiOpe1RyFHeI`

then we get a crash

```
java.lang.IllegalStateException: RouteStackManager: navigate target image_preview/https://fastly.picsum.photos/id/866/200/300.jpg?hmac=rcadCENKh4rD6MAp6V_ma-AyWv641M4iiOpe1RyFHeI not found
```

**Comment by FunkyMuse** [comment]
Workaround as of now

```kotlin
@Serializable
    data class ImagePreview(val url : ByteArray) : AppDestinations {

        val imageUrl get() = url.decodeToString()
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (other == null || this::class != other::class) return false

            other as ImagePreview

            return url.contentEquals(other.url)
        }

        override fun hashCode(): Int {
            return url.contentHashCode()
        }
    }
```

```kotlin
onImageClick = { imageUrl : String ->
                navigator.navigate(AppDestinations.ImagePreview(imageUrl.encodeToByteArray()))
            }
```

**Comment by Lavmee** [comment]
Hi, I'm working on it, I think it will be ready in 0.2.3

**Comment by Lavmee** [comment]
You can try the new version (0.2.3)?

**Comment by FunkyMuse** [comment]
it doesn't seem to work

when i pass the argument
`https://fastly.picsum.photos/id/866/200/300.jpg?hmac=rcadCENKh4rD6MAp6V_ma-AyWv641M4iiOpe1RyFHeI`

i receive

`httpshttps://fastly.picsum.photos/id/866/200/300.jpg?hmac=rcadCENKh4rD6MAp6V_ma-AyWv641M4iiOpe1RyFHeI`

**Comment by Lavmee** [comment]
Sorry, my fault, fix in version 0.2.4.

**Comment by FunkyMuse** [comment]
seems to be fixed, thanks, i'll do more thorough testing and reopen if I see a regression
